### PR TITLE
Removes or hides some types we don't want to commit to publicly

### PR DESCRIPTION
**Pull request by gustavoguichard**
This PR also fixes the bug where `collect` wouldn't accept inline cfs

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -114,20 +91,14 @@ type AllArguments<
...
-    ? never
-    : AllArguments<
-        Internal.RecordValuesFromKeysTuple<T, Internal.Keys<T>>
-      > extends ['Fail to compose', ...any[]]
-    ? AllArguments<Internal.RecordValuesFromKeysTuple<T, Internal.Keys<T>>>
-    : Internal.Zip<
-        Internal.Keys<T>,
-        AllArguments<Internal.RecordValuesFromKeysTuple<T, Internal.Keys<T>>>
-      >
+type CollectArguments<T extends Record<string, Composable>> = AllArguments<
```
Comment (diogob) [inline src/types.ts:94]: :raised_hands:

File: src/types.ts [hunk 2]
```diff
@@ -143,43 +114,33 @@ type SerializedResult<T> =
...
-      data: T
-    }
-  | { success: false; error: { issues: ParserIssue[] } }
-
 /**
  * The object used to validate either input or environment when creating domain functions.
  */
 type ParserSchema<T extends unknown = unknown> = {
-  safeParseAsync: (a: unknown) => Promise<ParserResult<T>>
+  safeParseAsync: (a: unknown) => Promise<
```
Comment (diogob) [inline src/types.ts:121]: much better!

**Review by diogob — COMMENTED** [review]
are the tests passing on your machine?
Here I got this:

```sh
❯ deno task test
Task test deno test --allow-env --allow-net src
...
error: TS2322 [ERROR]: Type 'Composable<(input?: unknown, environment?: unknown) => string>' is not assignable to type 'Composable<(input?: unknown, environment?: unknown) => string> & Composable<(a: number, b: number) => number>'.
  Type 'Composable<(input?: unknown, environment?: unknown) => string>' is not assignable to type 'Composable<(a: number, b: number) => number>'.
    Type 'Promise<Result<string>>' is not assignable to type 'Promise<Result<number>>'.
      Type 'Result<string>' is not assignable to type 'Result<number>'.
        Type 'Success<string>' is not assignable to type 'Result<number>'.
          Type 'Success<string>' is not assignable to type 'Success<number>'.
            Type 'string' is not assignable to type 'number'.
      str: withSchema(z.unknown(), z.any())((a) => String(a)),
      ~~~
    at file:///home/diogo/Projects/seasoned/composable-functions/src/tests/collect.test.ts:33:7

    The expected type comes from property 'str' which is declared here on type '{ add: Composable<(a: number, b: number) => number>; str: Composable<(input?: unknown, environment?: unknown) => string>; void: Composable<() => void>; } & { add: Composable<...>; str: Composable<...>; }'
          str: withSchema(z.unknown(), z.any())((a) => String(a)),
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        at file:///home/diogo/Projects/seasoned/composable-functions/src/tests/collect.test.ts:33:7
```

**Comment by gustavoguichard** [comment]
> are the tests passing on your machine? Here I got this:

It was, and also in CI, but any change in the code makes the problem go back... that collect with inline functions is a mistery!

**Pull request merged**